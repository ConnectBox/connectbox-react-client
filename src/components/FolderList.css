table.table tr a { display: block; }