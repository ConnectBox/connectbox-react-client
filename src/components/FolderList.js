import './FolderList.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import FolderListEntry from './FolderListEntry'

class FolderList extends Component {
    static propTypes = {
        content: PropTypes.array.isRequired,
        contentRoot: PropTypes.string.isRequired,
        folderPath: PropTypes.string.isRequired,
        iconMetadata: PropTypes.object.isRequired
    }
  //handles case 2 & 4 for top level and sub folder lists
  render () {
    const { content, contentRoot, folderPath, iconMetadata } = this.props

    return (
      <div className="table-responsive">
          <table className="table">
              <tbody>
                  {content.map((entry, i) =>
                    <FolderListEntry
                      contentRoot={contentRoot}
                      key={`folderEntry${i}`}
                      file={entry}
                      folderPath={folderPath}
                      iconMetadata={iconMetadata}
                  />)}
              </tbody>
          </table>
      </div>
    )
  }
}

export default FolderList
