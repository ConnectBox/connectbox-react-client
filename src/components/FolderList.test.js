import React from 'react'
import FolderList from './FolderList'
import renderer from 'react-test-renderer'

test('Render FolderList', () => {
  const iconMetadata = { ids: [], names: {}}
  const content = [{type: 'file', ext: 'doc', name: 'Test File'}]
  const component = renderer.create(
    <FolderList
      contentRoot='/content'
      content={content}
      folderPath='foo'
      iconMetadata={iconMetadata} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
