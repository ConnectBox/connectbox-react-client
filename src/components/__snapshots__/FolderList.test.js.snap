// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FolderList 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <a
            href="/contentfoo/Test File"
          >
            <i
              className="fa "
            />
            <i
              className="fa fa-file-text-o"
            />
            <span
              className=""
            >
              <span>
                Â 
                Test File
              </span>
            </span>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
